File: go/algorithms/easy/appleAndOrange.go
/*
Author: Rithie Natan Carvalhaes Prado
Date: 2026-07-13
Last modified: 2026-07-13
Description: Apple and Orange - HackerRank
See: https://www.hackerrank.com/challenges/apple-and-orange/problem
*/
//--------- package ---------
package easy


//--------- imports ---------
import (
	"fmt"
)


//--------- functions ---------
// Counts the number of apples and oranges that fall on Sam's house.
func countApplesAndOranges(s, t, a, b int32, apples, oranges []int32) {
	countApples := int32(0);
	countOranges := int32(0);

	for _, apple := range apples {
		reach := a + apple;
		if s <= reach && reach <= t {
			countApples++;
		}
	}

	for _, orange := range oranges {
		reach := b + orange;
		if s <= reach && reach <= t {
			countOranges++;
		}
	}

	fmt.Println(countApples);
	fmt.Println(countOranges);
}//end countApplesAndOranges()

File: java/algorithms/easy/appleAndOrange.java
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-07-13
 * Last modified: 2026-07-13
 * @description Apple and Orange - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/apple-and-orange/problem}
 */
//---------- Imports ----------
import java.util.*;


//---------- solution ----------
class Result 
{
    /**
     * Counts the number of apples and oranges that fall on Sam's house.
     * 
     * @param {int} s
     * @param {int} t
     * @param {int} a
     * @param {int} b
     * @param {int[]} apples
     * @param {int[]} oranges
     */
    public static void countApplesAndOranges(int s, int t, int a, int b, List<Integer> apples, List<Integer> oranges)
    {
        int countApples = 0;
        int countOranges = 0;

        for(int apple: apples)
        {
            int reach = a + apple;
            if(s <= reach && reach <= t)
            { countApples++; }
        }//end for

        for(int orange: oranges)
        {
            int reach = b + orange;
            if(s <= reach && reach <= t)
            { countOranges++; }
        }//end for

        System.out.println(countApples);
        System.out.println(countOranges);
    }//end countApplesAndOranges()
}//end class